### PR TITLE
RavenDB-7169 - Fix problem with returning "NoContent" to raw json request

### PR DESCRIPTION
**Pull request by maximburyak**

### CODE REVIEW

File: src/Raven.Client/Http/RavenCommand.cs [hunk 1]
```diff
@@ -84,7 +84,7 @@ public bool IsFailedWithNode(ServerNode node)
         {
             using (response)
             {
-                if (ResponseType == RavenCommandResponseType.Empty)
+                if (ResponseType == RavenCommandResponseType.Empty || response.StatusCode == HttpStatusCode.NoContent)
```
Comment (ayende) [inline src/Raven.Client/Http/RavenCommand.cs:87]: What caused it? Shouldn't the expected response match?

Comment (maximburyak) [reply src/Raven.Client/Http/RavenCommand.cs:87]: We are using ExecuteJsonAsync, which uses JsonCommandWithPayload, and it uses the default Object type. We could add JsonCommandWithPayload a parameter that will cause the RavenCommandResposeType to be using the right response type, but using the StatusCode for this matter looks safer.

Comment (ayende) [reply src/Raven.Client/Http/RavenCommand.cs:87]: Why are we using `JsonCommandWithPayload` if there is no expected payload

Comment (maximburyak) [reply src/Raven.Client/Http/RavenCommand.cs:87]: Actually, that class is part of the test infrastructure, meaning we could just add another class, there is another issue though: I think that the RavenCommand class should have an overload that receives a response type

**Pull request merged**